pallets/creditcoin/src/migrations/mod.rs

```diff
@@ -8,6 +8,24 @@ mod v4;
 mod v5;
 mod v6;
 
+#[cfg(feature = "try-runtime")]
+pub(crate) fn pre_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
+	// potentially do some checks on the state before the migration
+	if version < 6 {
+		v6::pre_upgrade::<T>()?;
+	}
+	Ok(())
+}
+
+#[cfg(feature = "try-runtime")]
+pub(crate) fn post_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
+	// do some post-migration checks
+	if version < 6 {
+		v6::post_upgrade::<T>()?;
+	}
+	Ok(())
+}
+
```

Comment:
The version checks look a little flaky. On paper, the branch will be executed for any previous upgrade which may not be a problem on any network but, now the function's correctness depends on the runtime state. Also, it becomes tough to reason about the assertion scope when the caller suggests that the call applies to any pre6 runtime state while the callee seems defined under v6 alone.

Reply:
Hmm it is a bit flaky, but I'm not sure how to make it better - any suggestions?

We could also improve this in a follow-up PR, since you've already approved this one